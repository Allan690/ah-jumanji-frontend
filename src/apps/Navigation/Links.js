import React from "react";

import NavItem from "./NavItem";
import NavButton from "./NavButton";
import DropDownItem from './NavDropDown';
import DropDownNotification from './NavDropDownNotification';
import defaultUserIcon from '../../assets/img/default-avatar.jpg';


const NotificationLink = () => (
  <React.Fragment>
    <button className="nav-link dropdown-toggle btn-link" type="button" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
      <span className="badge badge-danger notifications-badge">
        2
      </span>
      <i className="fas fa-bell" />
      Notifications
    </button>
  </React.Fragment>
);

export const HomeLinks = () => (
  <React.Fragment>
    <NavItem classnameOuter="nav-link active" classname="fas fa-home" link="/a/home" label="Home" />
    <NavItem classnameOuter="nav-link" classname="fas fa-search" data-toggle="modal" data-target="#search" link="javascript:;" label="Search" />
    <li className="nav-item dropdown">
      <NotificationLink />
      <ul className="dropdown-menu" id="dropDownNotifications" aria-labelledby="navbarDropdown">
        <DropDownNotification imgSrc={defaultUserIcon} userName="Edward Snowden" message="published an article" period="5 hrs" />
        <div className="dropdown-divider" />
        <DropDownNotification imgSrc={defaultUserIcon} userName="Edward Snowden" message="published an article" period="5 hrs" />
        <div className="dropdown-divider" />
      </ul>
    </li>
    <li className="nav-item dropdown">
      <button className="nav-link dropdown-toggle btn-link" type="button" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <img src={defaultUserIcon} alt="follower avatar" className="followers-avatar navbar-avatar rounded-circle" />
      </button>
      <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-user" link="/a/profile" label="Profile" />
        <div className="dropdown-divider" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-out-alt" link="/" label="Logout" />
      </ul>
    </li>
    <NavButton classnameOuter="nav-link" classname="btn btn-success btn-sm fas fa-plus-square" link="/a/createarticle" label="&nbsp;Add Post" />
  </React.Fragment>
);

export const HomeLinksSm = () => (
  <React.Fragment>
    <li className="nav-item dropdown sm-dropdown">
      <button className="nav-link dropdown-toggle btn-link" type="button" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <i className="fas fa-align-right" />
      </button>
      <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-plus-square" link="/a/createarticle" label="Add Post" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-plus-square" link="javascript:;" data-toggle="modal" data-target="#search" label="Search" />
        <li>
          <a href="/a/notifications" className="dropdown-item">
            <span className="badge badge-danger">
            5
            </span>
            <i className="fas fa-bell" />
            Notifications
          </a>
        </li>
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-user" link="/a/profile" label="Profile" />
        <div className="dropdown-divider" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-out-alt" link="/" label="Logout" />
      </ul>
    </li>
  </React.Fragment>
);

export const NotAuthenticatedLinks = () => (
  <React.Fragment>
    <NavItem classnameOuter="nav-link" classname="fas fa-info-circle" link="/about" label="About Us" />
    <NavItem classnameOuter="nav-link" classname="fas fa-comments" link="/contact" label="Contact Us" />
    <NavButton classnameOuter="nav-link" classname="btn btn-outline-warning btn-sm" link="/" label="Login" />
    <NavButton classnameOuter="nav-link" classname="btn btn-outline-success  btn-sm" link="/register" label="Signup" />
  </React.Fragment>
);

export const NotAuthenticatedLinksSm = () => (
  <React.Fragment>
    <li className="nav-item dropdown sm-dropdown">
      <button
        type="button" className="nav-link dropdown-toggle" id="navbarDropdown" data-toggle="dropdown"
        aria-haspopup="true" aria-expanded="false">
        <i className="fas fa-align-right" />
      </button>
      <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-info-circle" link="/about" label="About Us" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-comments" link="/contact" label="Contact Us" />
        <div className="dropdown-divider" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-in-alt" link="/" label="Login" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-in-alt" link="/register" label="Signup" />
      </ul>
    </li>
  </React.Fragment>
);

export const AuthenticatedLinks = () => (
  <React.Fragment>
    <NavItem classnameOuter="nav-link" classname="fas fa-home" link="/a/home" label="Home" />
    <li className="nav-item dropdown">
      <button type="button" className="nav-link dropdown-toggle btn-link" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <span className="badge badge-danger notifications-badge">
        2
        </span>
        <i className="fas fa-bell" />
        Notifications
      </button>
      <ul className="dropdown-menu" id="dropDownNotifications" aria-labelledby="navbarDropdown">
        <DropDownNotification imgSrc={defaultUserIcon} userName="Edward Snowden" message="published an article" period="5 hrs" />
        <div className="dropdown-divider" />
        <DropDownNotification imgSrc={defaultUserIcon} userName="Edward Snowden" message="published an article" period="5 hrs" />
        <div className="dropdown-divider" />
      </ul>
    </li>
    <li className="nav-item dropdown">
      <button className="nav-link dropdown-toggle btn-link" type="button" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <img src={defaultUserIcon} alt="follower avatar" className="followers-avatar navbar-avatar rounded-circle" />
      </button>
      <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-user" link="/a/profile" label="Profile" />
        <div className="dropdown-divider" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-out-alt" link="/" label="Logout" />
      </ul>
    </li>
  </React.Fragment>
);

export const AuthenticatedLinksSm = () => (
  <React.Fragment>
    <li className="nav-item dropdown sm-dropdown">
      <button className="nav-link dropdown-toggle" type="button" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <i className="fas fa-align-right" />
      </button>
      <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
        <li>
          <a href="/a/notifications" className="dropdown-item">
            <span className="badge badge-danger">
            5
            </span>
            <i className="fas fa-bell" />
            Notifications
          </a>
        </li>
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-user" link="/a/profile" label="Profile" />
        <div className="dropdown-divider" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-out-alt" link="/" label="Logout" />
      </ul>
    </li>
  </React.Fragment>
);
