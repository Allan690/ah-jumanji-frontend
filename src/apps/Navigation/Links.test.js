import React from 'react';

import Enzyme, { shallow, mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

import { MemoryRouter } from 'react-router-dom';
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { NotificationLink, HomeLinks, HomeLinksSm, NotAuthenticatedLinks, NotAuthenticatedLinksSm, AuthenticatedLinks, AuthenticatedLinksSm} from './Links';

Enzyme.configure({ adapter: new Adapter() });
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('<NotificationLink />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<NotificationLink debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('NotificationLink', () => {
  it('should render self and subcomponents', () => {
    const store = mockStore({});
    const enzymeWrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <NotificationLink />
        </MemoryRouter>
      </Provider>,);
    expect(enzymeWrapper.find('NotificationLink')).toBeDefined();
  });
});

describe('<HomeLinks />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<HomeLinks debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('<HomeLinksSm />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<HomeLinksSm debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('HomeLinksSm', () => {
  it('should render self and subcomponents', () => {
    const store = mockStore({});
    const enzymeWrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <HomeLinksSm />
        </MemoryRouter>
      </Provider>,);
    expect(enzymeWrapper.find('HomeLinksSm')).toBeDefined();
  });
});

describe('<NotAuthenticatedLinks />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<NotAuthenticatedLinks debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('NotAuthenticatedLinks', () => {
  it('should render self and subcomponents', () => {
    const store = mockStore({});
    const enzymeWrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <NotAuthenticatedLinks />
        </MemoryRouter>
      </Provider>,);
    expect(enzymeWrapper.find('NotAuthenticatedLinks')).toBeDefined();
  });
});

describe('<NotAuthenticatedLinksSm />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<NotAuthenticatedLinksSm debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('NotAuthenticatedLinksSm', () => {
  it('should render self and subcomponents', () => {
    const store = mockStore({});
    const enzymeWrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <NotAuthenticatedLinksSm />
        </MemoryRouter>
      </Provider>,);
    expect(enzymeWrapper.find('NotAuthenticatedLinksSm')).toBeDefined();
  });
});

describe('<AuthenticatedLinks />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<AuthenticatedLinks debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('AuthenticatedLinks', () => {
  it('should render self and subcomponents', () => {
    const store = mockStore({});
    const enzymeWrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <AuthenticatedLinks />
        </MemoryRouter>
      </Provider>,);
    expect(enzymeWrapper.find('AuthenticatedLinks')).toBeDefined();
  });
});

describe('<AuthenticatedLinksSm />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<AuthenticatedLinksSm debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('AuthenticatedLinksSm', () => {
  it('should render self and subcomponents', () => {
    const store = mockStore({});
    const enzymeWrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <AuthenticatedLinksSm />
        </MemoryRouter>
      </Provider>,);
    expect(enzymeWrapper.find('AuthenticatedLinksSm')).toBeDefined();
  });
});
