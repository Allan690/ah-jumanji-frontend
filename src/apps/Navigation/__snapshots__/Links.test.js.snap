// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthenticatedLinks /> should match the snapshot 1`] = `
<Fragment>
  <NavItem
    classname="fas fa-home"
    classnameOuter="nav-link"
    label="Home"
    link="/a/home"
  />
  <li
    className="nav-item dropdown"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="nav-link dropdown-toggle btn-link"
      data-toggle="dropdown"
      id="navbarDropdown"
      type="button"
    >
      <img
        alt="follower avatar"
        className="followers-avatar navbar-avatar rounded-circle"
      />
    </button>
    <ul
      aria-labelledby="navbarDropdown"
      className="dropdown-menu"
    >
      <DropDownItem
        classname="fas fa-user"
        classnameOuter="dropdown-item"
        label="Profile"
        link="/a/profile"
      />
      <div
        className="dropdown-divider"
      />
      <DropDownItem
        classname="fas fa-sign-out-alt"
        classnameOuter="dropdown-item"
        label="Logout"
        link="/"
      />
    </ul>
  </li>
</Fragment>
`;

exports[`<AuthenticatedLinksSm /> should match the snapshot 1`] = `
<Fragment>
  <li
    className="nav-item dropdown sm-dropdown"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="nav-link dropdown-toggle"
      data-toggle="dropdown"
      id="navbarDropdown"
      type="button"
    >
      <i
        className="fas fa-align-right"
      />
    </button>
    <ul
      aria-labelledby="navbarDropdown"
      className="dropdown-menu"
    >
      <li>
        <a
          className="dropdown-item"
          href="/a/notifications"
        >
          <i
            className="fas fa-bell"
          />
          Notifications
        </a>
      </li>
      <DropDownItem
        classname="fas fa-user"
        classnameOuter="dropdown-item"
        label="Profile"
        link="/a/profile"
      />
      <div
        className="dropdown-divider"
      />
      <DropDownItem
        classname="fas fa-sign-out-alt"
        classnameOuter="dropdown-item"
        label="Logout"
        link="/"
      />
    </ul>
  </li>
</Fragment>
`;

exports[`<HomeLinks /> should match the snapshot 1`] = `
<Fragment>
  <NavItem
    classname="fas fa-home"
    classnameOuter="nav-link active"
    label="Home"
    link="/a/home"
  />
  <NavItem
    classname="fas fa-search"
    classnameOuter="nav-link"
    data-target="#search"
    data-toggle="modal"
    label="Search"
    link="javascript:;"
  />
  <NotificationLink />
  <li
    className="nav-item dropdown"
  >
    <ProfPic />
    <ul
      aria-labelledby="navbarDropdown"
      className="dropdown-menu"
    >
      <DropDownItem
        classname="fas fa-user"
        classnameOuter="dropdown-item"
        label="Profile"
        link="/a/profile"
      />
      <div
        className="dropdown-divider"
      />
      <DropDownItem
        classname="fas fa-sign-out-alt"
        classnameOuter="dropdown-item"
        label="Logout"
        link="/"
      />
    </ul>
  </li>
  <NavButton
    classname="btn btn-success btn-sm fas fa-plus-square"
    classnameOuter="nav-link"
    label="Â Add Post"
    link="/a/createarticle"
  />
</Fragment>
`;

exports[`<HomeLinksSm /> should match the snapshot 1`] = `
<Fragment>
  <li
    className="nav-item dropdown sm-dropdown"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="nav-link dropdown-toggle btn-link"
      data-toggle="dropdown"
      id="navbarDropdown"
      type="button"
    >
      <i
        className="fas fa-align-right"
      />
    </button>
    <ul
      aria-labelledby="navbarDropdown"
      className="dropdown-menu"
    >
      <DropDownItem
        classname="fas fa-plus-square"
        classnameOuter="dropdown-item"
        label="Add Post"
        link="/a/createarticle"
      />
      <DropDownItem
        classname="fas fa-plus-square"
        classnameOuter="dropdown-item"
        data-target="#search"
        data-toggle="modal"
        label="Search"
        link="javascript:;"
      />
      <li>
        <a
          className="dropdown-item"
          href="/a/notifications"
        >
          <i
            className="fas fa-bell"
          />
          Notifications
        </a>
      </li>
      <DropDownItem
        classname="fas fa-user"
        classnameOuter="dropdown-item"
        label="Profile"
        link="/a/profile"
      />
      <div
        className="dropdown-divider"
      />
      <DropDownItem
        classname="fas fa-sign-out-alt"
        classnameOuter="dropdown-item"
        label="Logout"
        link="/"
      />
    </ul>
  </li>
</Fragment>
`;

exports[`<NotAuthenticatedLinks /> should match the snapshot 1`] = `
<Fragment>
  <NavItem
    classname="fas fa-info-circle"
    classnameOuter="nav-link"
    label="About Us"
    link="/about"
  />
  <NavItem
    classname="fas fa-comments"
    classnameOuter="nav-link"
    label="Contact Us"
    link="/contact"
  />
  <NavButton
    classname="btn btn-outline-warning btn-sm"
    classnameOuter="nav-link"
    label="Login"
    link="/"
  />
  <NavButton
    classname="btn btn-outline-success  btn-sm"
    classnameOuter="nav-link"
    label="Signup"
    link="/register"
  />
</Fragment>
`;

exports[`<NotAuthenticatedLinksSm /> should match the snapshot 1`] = `
<Fragment>
  <li
    className="nav-item dropdown sm-dropdown"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="nav-link dropdown-toggle"
      data-toggle="dropdown"
      id="navbarDropdown"
      type="button"
    >
      <i
        className="fas fa-align-right"
      />
    </button>
    <ul
      aria-labelledby="navbarDropdown"
      className="dropdown-menu"
    >
      <DropDownItem
        classname="fas fa-info-circle"
        classnameOuter="dropdown-item"
        label="About Us"
        link="/about"
      />
      <DropDownItem
        classname="fas fa-comments"
        classnameOuter="dropdown-item"
        label="Contact Us"
        link="/contact"
      />
      <div
        className="dropdown-divider"
      />
      <DropDownItem
        classname="fas fa-sign-in-alt"
        classnameOuter="dropdown-item"
        label="Login"
        link="/"
      />
      <DropDownItem
        classname="fas fa-sign-in-alt"
        classnameOuter="dropdown-item"
        label="Signup"
        link="/register"
      />
    </ul>
  </li>
</Fragment>
`;

exports[`<NotificationLink /> should match the snapshot 1`] = `""`;
